File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.go
func vowelStrings(words []string, queries [][]int) []int {
    var size = len(words)
    var vowelCount = make([]int, size)
    var vowels = make(map[byte]bool)
    for _, c := range "aeiou" { vowels[byte(c)] = true }
    
    var prev = 0
    for i := range size {
        if vowels[words[i][0]] && vowels[words[i][len(words[i]) - 1]] {
            vowelCount[i] += 1
        }
        vowelCount[i] += prev
        prev = vowelCount[i]
    }
    
    var ret = []int{}
    for _, query := range queries {
        var ans = vowelCount[query[1]]
        if query[0] > 0 {
            ans -= vowelCount[query[0] - 1]
        }
        ret = append(ret, ans)
    }
    return ret
}

File: DCP-01-25/2559-Count-Vowel-Strings-in-Ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        vowels = set(['a', 'e', 'i', 'o', 'u'])
        res = [1 if (words[0][0] in vowels and words[0][-1] in vowels) else 0]
        for i in range(1, len(words)):
            res.append(res[-1] + (words[i][0] in vowels and words[i][-1] in vowels))
        
        print(res)
        ret = []
        for a, b in queries:
            s = res[b]
            if a > 0:
                s -= res[a - 1]
            
            ret.append(s)
        
        return ret
            
            
